Iso installer (#147)
SUSE/elemental: docs/linux-only-image.md

        elemental3-toolkit version
        ```

## Create an Installer ISO

In order to create a self installer ISO some configuration assets must be prepared in advance to include them into the ISO. The essential parts are:

1. A configuration script
2. A directory tree to include additional files into the ISO filesystem

### Configure the Live Installer

The ISO supports configurations through a script which will be executed in late initramfs just before switching root.
The script is executed chrooting to the system root.

#### Example Live Config Script

In this example we are going to setup a configuration script that will set three aspects:

1. Autologin, so the live ISO does not require a root password
2. Add an elemental-autoinstaller service to run the installation at boot

> 3. Ensure the extensions added in the ISO filesystem are linked to `/run/extesions` so they are loaded at boot

Reviewer: Seems I cannot make suggestions here, but there's a small typo here: `/run/extensions` note the missing "n".

a systemd extension so to make it available at ISO boot it has to be present in the ISO filesystem and copied or linked to `/run/extensions`.

This example shows how to prepare the ISO overlay directory tree and the configuration script to ensure the `elemental3-toolkit` extensions are
available and loaded at boot.

1. Create an `iso-overlay/extensions` directory:

    ```shell
    mkdir -p iso-overlay/extensions
    ```

2. Move the `elemental3-toolkit-3.0.x86-64.raw` extension image to this directory:

    ```shell
    mv example-extension/mkosi.output/elemental3-toolkit-3.0.x86-64.raw iso-overlay/extensions
    ```

3. Make sure the live configuration script links the `extensions` folder at `/run/extensions`

### Build the Installer ISO

To create an installer live ISO image process is similar to the [install step](#install-os-on-target-device) described above. The command below
will create an ISO image inside the `build` output folder using an `openSUSE Tumbleweed` image and it will be configured to automatically self
install to the target device (e.g. `/dev/sda`) at boot.

```shell
sudo elemental3-toolkit --debug build-iso \
    --output build \

>     --os-image registry.opensuse.org/devel/unifiedcore/tumbleweed/containers/uc-base-os-kernel-default:latest \

Author: Yes indeed, this requires an update gonna make a submission

>     --os-image registry.opensuse.org/devel/unifiedcore/tumbleweed/containers/uc-base-os-kernel-default:latest \

Author: submission done at https://build.opensuse.org/requests/1296048

    --overlay dir://iso-overlay \
    --config config-live.sh \
    --install-target /dev/sda \
    --install-overlay tar://overlays.tar.gz \
    --install-config config.sh
```

Note that:
* The `overlays.tar.gz` tarball came from the system extension image [example configuration](#example-system-extension-image).
* The `config.sh` script came from the [configuration script example](#example-config-script).
* The `/dev/sda` is the target device you want the ISO to install to.
* The `iso-overlay` is the directory tree [including extensions](#include-extensions-in-the-installer-media) that will be included in the ISO filesystem of the built image.
* The `config-live.sh` script came from the live [configuration script example](#example-live-config-script).

### Booting a Live Installer Image

In order to boot the installer ISO and test the automatic install a simple QEMU virtual machine can be launched with the following command:

```shell

> qemu-kvm -m 8190 \

Author: We could  add it yes, even it feels like an opinionated choice.

        elemental3-toolkit version
        ```

## Create an Installer ISO

> To create a self installer ISO, users must prepare and include a specific set of configuration assets. These include:

Reviewer 2: ```suggestion
To create a self installer ISO, you should prepare and include a specific set of configuration assets. These include:
```
